### ISSUE TITLE
Accessing undefined variable Proto, ProtoField, DissectorTable, base

### ISSUE THREAD

**Issue by shakthimaan** [mpeterv/luacheck#144]
I am writing a Wireshark Lua plugin, and I directly use Proto, ProtoField, DissectorTable and base in the code. The plugin works with Wireshark, but, Luacheck warns me with the following:
```
accessing undefined variable Proto
accessing undefined variable ProtoField
accessing undefined variable base
accessing undefined variable DissectorTable
```
The above are available as part of Wireshark LuaAPI:
https://wiki.wireshark.org/LuaAPI/Proto

How can I fix the warnings?

**Comment by mpeterv** [comment]
You have to allow working with these custom globals. E.g. add `.luacheckrc` file with line:

```lua
globals = {"Proto", "ProtoField", "base", "DissectorTable"}
```

This allows both accessing and mutating the globals. Use `read_globals` instead of `globals` to allow only accessing. There is also a way to control which fields of these globals can be used: [docs](https://luacheck.readthedocs.io/en/stable/config.html#custom-sets-of-globals).

**Comment by shakthimaan** [comment]
Used read_globals. Thanks!
